packages/jellyfish-api-core/src/category/masternode.ts

```diff
@@ -0,0 +1,142 @@
+   * @param {CreateMasternodeOptions} [options]
+   * @param {UTXO[]} [options.utxos = []]
+   * @param {string} [options.uxtos.txid] The transaction id
+   * @param {string} [options.utxos.vout] The output number
+   * @return {Promise<string>}
+   */
+  async createMasternode (
+    ownerAddress: string,
+    operatorAddress?: string,
+    options: CreateMasternodeOptions = { utxos: [] }
+  ): Promise<string> {
+    operatorAddress = operatorAddress ?? ownerAddress
+    return await this.client.call('createmasternode', [ownerAddress, operatorAddress, options.utxos], 'number')
+  }
+
+  /**
+   * Returns information about multiple masternodes.
+   *
+   * @param {MasternodePagination} pagination
+   * @param {string} [pagination.start]
+   * @param {boolean} [pagination.including_start = true] Include starting position.
+   * @param {string} [pagination.limit = 100] Maximum number of orders to return.
+   * @param {boolean} [verbose = true] Flag for verbose list. Only ids are returned when false.
+   * @return {Promise<MasternodeResult | string>}
+   */
+  listMasternodes (pagination?: MasternodePagination, verbose?: boolean): Promise<MasternodeResult>
+
+  /**
+   * Returns information about multiple masternodes.
+   *
+   * @param {MasternodePagination} pagination
+   * @param {string} [pagination.start]
+   * @param {boolean} [pagination.including_start = true] Include starting position.
+   * @param {string} [pagination.limit = 100] Maximum number of orders to return.
+   * @param {boolean} verbose true
+   * @return {Promise<MasternodeResult | string>}
+   */
+  listMasternodes (pagination: MasternodePagination, verbose: true): Promise<MasternodeResult>
+
+  /**
+   * Returns information about multiple masternodes.
+   *
+   * @param {MasternodePagination} pagination
+   * @param {string} [pagination.start]
+   * @param {boolean} [pagination.including_start = true] Include starting position.
+   * @param {string} [pagination.limit = 100] Maximum number of orders to return.
+   * @param {boolean} verbose false.
+   * @return {Promise<MasternodeResult | string>}
+   */
+  listMasternodes (pagination: MasternodePagination, verbose: false): Promise<MasternodeResult>
+
+  /**
+   * Returns information about multiple masternodes.
+   *
+   * @param {MasternodePagination} pagination
+   * @param {string} [pagination.start]
+   * @param {boolean} [pagination.including_start = true] Include starting position.
+   * @param {string} [pagination.limit = 100] Maximum number of orders to return.
+   * @param {boolean} [verbose = true] Flag for verbose list. Only ids are returned when false.
+   * @return {Promise<MasternodeResult | string>}
+   */
+  async listMasternodes (
+    pagination: MasternodePagination = {
+      including_start: true,
+      limit: 100
+    },
+    verbose: boolean = true
+  ): Promise<MasternodeResult> {
+    return await this.client.call('listmasternodes', [pagination, verbose], 'number')
+  }
```

Comment:
You need to handle multiple types coming out of the RPC. Hence you need to setup generic with different verbose. 

Look at this for reference: https://github.com/DeFiCh/jellyfish/blob/bf3e5bb2853ed2fff12f65c52deee4d16e562dd8/packages/jellyfish-api-core/src/category/blockchain.ts#L42-L77